### PR TITLE
REFACTOR-#7008: Remove `check_exception_type` argument of `eval_general` function

### LINKED ISSUE #7008: Remove `check_exception_type` argument of `eval_general` function
Using it not only eliminates exception checking, but also implicitly eliminates running on modin, this approach is error prone.

### PR DESCRIPTION
**Pull request by anmyachev**
## What do these changes do?

- [x] first commit message and PR title follow format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
  > **_NOTE:_**  If you edit the PR title to match this format, you need to add another commit (even if it's empty) or amend your last commit for the CI job that checks the PR title to pick up the new PR title.
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #7008 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date

### CODE REVIEW

File: modin/pandas/test/dataframe/test_default.py [hunk 1]
```diff
@@ -643,7 +655,22 @@ def test_pivot(data, index, columns, values):
         pytest.param("nunique", id="full_axis_func"),
     ],
 )
-def test_pivot_table_data(data, index, columns, values, aggfunc):
+def test_pivot_table_data(data, index, columns, values, aggfunc, request):
+    if (
```
Comment (anmyachev) [inline modin/pandas/test/dataframe/test_default.py:659]: It’s not pretty, but it’s difficult to select the desired condition through parameters. It is also possible that there are several problems that will need to be investigated separately.

File: modin/pandas/test/dataframe/test_default.py [hunk 2]
```diff
@@ -585,20 +586,35 @@ def melt(df, *args, **kwargs):
 
 @pytest.mark.parametrize("data", test_data_values, ids=test_data_keys)
 @pytest.mark.parametrize(
-    "index", [lambda df: df.columns[0], lambda df: df[df.columns[0]].values, None]
+    "index",
+    [lambda df: df.columns[0], lambda df: df.columns[:2], lib.no_default],
+    ids=["one_column_index", "several_columns_index", "default"],
```
Comment (YarShev) [inline modin/pandas/test/dataframe/test_default.py:591]:
```suggestion
    ids=["one_column_index", "several_columns_index", "no_default"],
```
?

File: modin/pandas/test/dataframe/test_default.py [hunk 3]
```diff
@@ -585,20 +586,35 @@ def melt(df, *args, **kwargs):
...
+)
+@pytest.mark.parametrize(
+    "columns", [lambda df: df.columns[len(df.columns) // 2]], ids=["one_column"]
```
Comment (YarShev) [inline modin/pandas/test/dataframe/test_default.py:594]: Why one column? We pick columns by len(df.columns) // 2?

File: modin/pandas/test/dataframe/test_default.py [hunk 4]
```diff
@@ -585,20 +586,35 @@ def melt(df, *args, **kwargs):
...
 )
-@pytest.mark.parametrize("columns", [lambda df: df.columns[len(df.columns) // 2]])
 @pytest.mark.parametrize(
-    "values", [lambda df: df.columns[-1], lambda df: df.columns[-2:], None]
+    "values",
+    [lambda df: df.columns[-1], lambda df: df.columns[-2:], lib.no_default],
+    ids=["one_column_values", "several_columns_values", "default"],
```
Comment (YarShev) [inline modin/pandas/test/dataframe/test_default.py:599]:
```suggestion
    ids=["one_column_values", "several_columns_values", "no_default"],
```
?

File: modin/pandas/test/dataframe/test_default.py (hunk 3 above)
Comment (anmyachev) [reply modin/pandas/test/dataframe/test_default.py:594]: We take one column from the middle, there is no slice right there.

Comment (YarShev) [reply modin/pandas/test/dataframe/test_default.py:594]: Ah, I see. Missed that.

File: modin/pandas/test/dataframe/test_default.py (hunk 2 above)
Comment (YarShev) [reply modin/pandas/test/dataframe/test_default.py:591]: makes sense

**Review by YarShev — APPROVED** [review]

**Pull request merged**